core-bundle/tests/Doctrine/Backup/DumperTest.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Doctrine\Backup;
+
+use Contao\CoreBundle\Doctrine\Backup\Backup;
+use Contao\CoreBundle\Doctrine\Backup\BackupManagerException;
+use Contao\CoreBundle\Doctrine\Backup\Config\CreateConfig;
+use Contao\CoreBundle\Doctrine\Backup\Dumper;
+use Contao\TestCase\ContaoTestCase;
+use Doctrine\DBAL\Cache\ArrayResult;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Result;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Schema\View;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\DBAL\Types\Types;
+
+class DumperTest extends ContaoTestCase
+{
+    /**
+     * @dataProvider successfulDumpProvider
+     */
+    public function testSuccessfulDump(array $tables, array $views, array $queries, array $expectedDump): void
+    {
+        $backup = new Backup('backup__20211101141254.sql');
+
+        $connection = $this->mockConnection($tables, $views, $queries);
+        $dumper = new Dumper();
+        $config = (new CreateConfig($backup));
+
+        $this->assertSame($expectedDump, iterator_to_array($dumper->dump($connection, $config), false));
+    }
+
+    public function testUnsuccessfulDump(): void
+    {
+        $this->expectException(BackupManagerException::class);
+        $this->expectExceptionMessage('Error!');
+
+        $backup = new Backup('backup__20211101141254.sql');
+
+        $connection = $this->createMock(Connection::class);
+        $connection
+            ->expects($this->once())
+            ->method('createSchemaManager')
+            ->willThrowException(new \Exception('Error!'))
+        ;
+
+        $dumper = new Dumper();
+        $config = (new CreateConfig($backup))->withGzCompression(false);
+
+        iterator_to_array($dumper->dump($connection, $config), false);
```

Comment:
<img width="947" alt="" src="https://user-images.githubusercontent.com/1192057/143050926-cb300bb0-8edc-4d90-80be-52d5b00cd678.png">

Reply:
That's because of your CS changes. You changed `@return static` to `self`. `self` is the same class (= `AbstractConfig`). But `static` would be correct. Which is natively only available as of PHP 8, see https://wiki.php.net/rfc/static_return_type